anoncreds/src/services/verifier.rs

```diff
@@ -355,6 +363,10 @@ fn compare_timestamps_from_proof_and_request(
         })
         .collect::<Result<Vec<()>>>()?;
 
+    if !warnings.is_empty() {
+        nrp_warnings.timestamps_out_of_range = Some(warnings);
+    }
+
```

Comment:
Would it be worth it to instantiate a new one, shadowing the previous one, and keeping `timestamps_out_of_range` private? Not sure if it would work.